### PR TITLE
Mandatorily Make Projects Public After HydroShare Export

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

When a project is exported to HydroShare, there may be a link from HydroShare back to the project in MMW. To allow this, we need to make the project public in MMW so that any incoming user can see it.

After a successful export to HydroShare, we update the project's privacy and make it unchangeable from the MultiShareView. We also add a message explaining that any project exported to HydroShare will always have link sharing enabled.

Connects #2638 

### Demo

Once a project has been exported to HydroShare, the modal looks like this:

image

For a project not exported to HydroShare, the link sharing can be toggled independently:

image

Turning HydroShare off leaves Link Sharing on, but can be toggled independently by the user.

Tagging @ajrobbins and @jfrankl for visual review.

## Testing Instructions

* Check out this branch and `bundle`
* Ensure your VM is setup to work with HydroShare. This essentially involves `vagrant ssh app -c 'sudo vim /etc/mmw.d/env/MMW_HYDROSHARE_SECRET_KEY'` and entering the secret key for HydroShare Beta from LastPass. Restart `mmw-app` after changing that file.
* Go to [:8000/](http://localhost:8000/) and create a project
* Export it to HydroShare. Ensure that if the export fails or is canceled, link sharing remains off.
* Ensure that a successful export turns link sharing off, and that you are unable to turn it off.
* Ensure that removing a project from HydroShare leaves link sharing on, but can be turned off by the user.

### CODE REVIEW

**Comment by ajrobbins** [comment]
LGTM visually!

**Comment by arottersman** [comment]
Taking a look!

File: src/mmw/js/src/modeling/models.js [hunk 1]
```diff
@@ -464,8 +464,11 @@ var ProjectModel = Backbone.Model.extend({
                 mapshed_data: mapshedData,
             }, payload))
         }).done(function(result) {
-            self.set('hydroshare', result);
-            self.set('hydroshare_errors', []);
+            self.set({
+                hydroshare: result,
+                hydroshare_errors: [],
+                is_private: false,
```
Comment (arottersman) [inline src/mmw/js/src/modeling/models.js]: This is a little unconventional in its assumption that the server is also updating this field, but I understand that receiving the updated project from the server to kick off this model change might require some refactoring that it's maybe not worth investing in now.

Comment (rajadain) [reply src/mmw/js/src/modeling/models.js]: Agreed. While a refactoring would be expensive, I will add a comment for clarity.

**Comment by rajadain** [comment]
Added a comment to the effect, explaining the behavior pointed out in https://github.com/WikiWatershed/model-my-watershed/pull/2658#discussion_r168302959. Will merge once green.

Thanks for taking a look! ⚖️

**Comment by jfrankl** [comment]
Looks good

**Pull request merged**